Support for "bitsets" (and bitfields)

otobrglez opened the pull request:
Hello! 👋 

As per some brief discussion in [this issue](https://github.com/profunktor/redis4cats/issues/604) I took the challenge of adding a few Redis commands to redis4cats.

At the moment this PR is still in work-in-progress mode as I have a few questions that we can hopefully resolve before this gets merged.

1. Should I propose this PR against `master` or against any other branch?
2. Although it was recommended that bitmaps should be added to `BitCommands` trait; some methods belong more to `Setters` trait upon my investigation.  😅  Any guidance on that might be welcomed. I did however add method `setBit` there now.
3. The bitfield command needs to be built as a wrapper on top of Lettuce where such interaction is done with the help of [`BitFieldArgs`](https://lettuce.io/core/5.0.0.M2/api/io/lettuce/core/BitFieldArgs.html) - for [`bitfield`](https://lettuce.io/core/5.0.0.M2/api/io/lettuce/core/api/sync/RedisStringCommands.html#bitfield-K-io.lettuce.core.BitFieldArgs-). So I'm wondering how should that abstraction really be made so that it fits well within the design of this lib and Scala ecosystem as such. In the early spike this as seen here you'll see that I thought about using tuples... and ideas there?

I would appreciate some feedback and guidance so that I can finish this.

Thank you and have a nice day!

gvolpe commented:
Hi @otobrglez , thanks for the submission! I'll be looking into it shortly, just catching up as I just came back from holidays.

otobrglez commented:
Hey @gvolpe 👋 

These are the changes that should hopefully reflect the discussion and proposals you've made in the discussion above.

1. Existing "Bitmap" functions were moved to the `BitCommands` trait that lives inside `bitmaps.scala` file. After that the `Bits` trait became obsolete and I removed it.
2. `redis.scala` was refactored so that bitmap operations now live together - similar to how list and sets are "together".
3. `bitfield` command was added to `BitCommands`
4. Example file for bitmaps was created
5. I've added tests for bitmaps to test suite.

Now, this client is only missing [`BITFIELD_RO`](https://redis.io/commands/bitfield_ro) command support; but in other to get that - we probably need to patch the Lettuce for that. 👼 

Please give this another look and let me know if you wanna change something etc..

P.s.: No rush as I need to finish this book that I got in my mail today. ✌️ 

![image](https://user-images.githubusercontent.com/225946/136983053-4be4d28d-e96d-4256-b1d4-c94d98546457.png)

gvolpe commented:
@otobrglez , thanks a lot, I was writing at the same time.

Feel free to create an issue for that BITFIELD_RO, so we don't forget to add it when Lettuce is updated.

Awesome! Hope you enjoy the book 😃
